linker/shared/src/test/scala/org/scalajs/linker/LinkerTest.scala

```diff
@@ -50,6 +52,39 @@ class LinkerTest {
     testLink(helloWorldClassDefs, MainTestModuleInitializers)
   }
 
+  /** Makes sure that linking a "substantial" program (using `println`) does
+   *  not trigger the second attempt in the Emitter due to dangerous global
+   *  refs.
+   */
+  @Test
+  def linkNoSecondAttemptInEmitter(): AsyncResult = await {
+    val PrintStreamClass = ClassName("java.io.PrintStream")
+    val PrintStreamType = ClassType(PrintStreamClass)
+
+    val classDefs = List(
+      mainTestClassDef {
+        val err = ApplyStatic(EAF, "java.lang.System",
+            m("err", Nil, ClassRef(PrintStreamClass)), Nil)(PrintStreamType)
+        Apply(EAF, err, m("println", List(O), V), List(str("Hello world!")))(NoType)
+      }
+    )
+
+    val logger = new CapturingLogger
+
+    val linker = StandardImpl.linker(StandardConfig())
+    val classDefsFiles = classDefs.map(MemClassDefIRFile(_))
+
+    for {
+      fulllib <- TestIRRepo.fulllib
+      report <- linker.link(fulllib ++ classDefsFiles,
+          MainTestModuleInitializers, MemOutputDirectory(), logger)
+    } yield {
+      val log = logger.allLogLines
+      assertFalse("The second attempt of the emitter should not be triggered\n" + log,
+          log.contains("Emitter: The set of dangerous global refs has changed."))
```

Comment:
Consider adding a test that tests this error message is present if a dangerous global ref is present. That way, we avoid the test becoming useless (without just relying on the comment).